### ISSUE TITLE
Support Angular by making an ES module

### ISSUE THREAD

**Issue by paviad** [gvergnaud/ts-pattern#53]
# Problem

When importing `ts-pattern` in a modern Angular app (tested on 12+), angular complains that:

```
Warning: <module> depends on 'ts-pattern'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies
```

# Desired Solution

Create an es-module version of this library.

# Alternatives

I'm aware of no alternatives.

**Comment by m-rutter** [comment]
I've commented on the likely right approach for solving this in your PR, but in terms of your particular usecase you can probably safely ignore this warning as its a tiny 1.4kb library, so any optimisations that the tooling you are using could apply would be negligible.

**Comment by craigmiller160** [comment]
I'll just chime in that it should be possible to run a build that produces two outputs, a commonjs and an esmodule. As long as the import patterns are the same, minus a directory prefix (ie, esmodule ones need "es" as the first directory in the import path), there's a Babel Transform Imports Plugin that will automatically translate the imports to make everything use esmodules. Likewise, you could also add an eslint rule that prevents importing without the "es" directory prefix.

Basically, once this library produces its dual-output, then it should be possible to easily use it as either commonjs or esmodules in any project.

PS. Please keep commonjs as the default for maximum out-of-the-box support. While I agree the whole ecosystem should move to esmodules, we're not there yet.

**Comment by gvergnaud** [comment]
Closing since TS-Pattern v4 has been released and now generates compiles the code for both commonjs and esmodules. [Release note](https://github.com/gvergnaud/ts-pattern/releases/tag/v4.0.1)
